docs/source/flow-project.rst

```diff
@@ -49,34 +49,48 @@ Defining a workflow
 
 We will reproduce the simple workflow introduced in the previous section by first copying both the ``greeted()`` condition function and the ``hello()`` *operation* function into the ``project.py`` module.
 We then use the :py:func:`~flow.FlowProject.operation` and the :py:func:`~.flow.FlowProject.post` decorator functions to specify that the ``hello()`` operation function is part of our workflow and that it should only be executed if the ``greeted()`` condition is not met.
+The :py:func:`~flow.FlowProject.make_group` is used to allow operations in ``FlowProject`` to be organized into *metaoperations*.
 
 .. code-block:: python
 
     # project.py
     from flow import FlowProject
+    
+    stranger_group = FlowProject.make_group(name='stranger',options='--num-passes=5')
+    friend_group = FlowProject.make_group(name='friend',options='--num-passes=3')
 
 
     class Project(FlowProject):
         pass
-
+        
+    def handshaked(job)
+        return job.isfile('shake_hand.txt')
 
     def greeted(job)
         return job.isfile('hello.txt')
-
-
+    
+ 
     @Project.operation
+    @stranger_group
     @Project.post(greeted)
     def hello(job):
         with job:
             with open('hello.txt', 'w') as file:
                 file.write('world!\n')
-
+                
+    @friend_group
+    @FlowProject.pre.after(stranger_group)
+    @FlowProject.operation
+    def shake(job):
+        with job:
+            with open('shake_hand.txt','w') as file:
+                file.write('Make a friendship!\n')
 
     if __name__ == '__main__':
         Project().main()
 
 We can define both *pre* and *post* conditions, which allow us to define arbitrary workflows as an acyclic graph.
-A operation is only executed if **all** pre-conditions are met, and at *at least one* post-condition is not met.
+A operation is only executed if **all** pre-conditions are met, and at *at least one* post-condition is not met. In addition, groups from :py:func:`~flow.FlowProject.make_group` are egliable for *pre* and *post* conditions.
```

Comment:
Technically if post conditions are bound "closer to the function" than group decorator then you can use `pre.after` otherwise, it does not work. Also groups do not have their own pre and post conditions. Rather preconditions are at least one operation in the group is eligible and post conditions are that all operation post conditions are met.